src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -898,18 +870,22 @@ export default class FilingHistoryList extends Vue {
 
   /** Whether to disable correction for this history item. */
   protected disableCorrection (item: HistoryItemIF): boolean {
-    return (
-      !this.isAllowed(AllowableActions.FILE_CORRECTION) ||
-      item.availableOnPaperOnly ||
-      item.isTypeStaff ||
-      item.isFutureEffective ||
-      this.isTypeAlteration(item) ||
-      this.isTypeTransition(item) ||
-      // the following corrections are supported:
-      (this.isTypeIncorporationApplication(item) && !this.isBComp) ||
-      (this.isTypeChangeOfRegistration(item) && !this.isFirm) ||
-      (this.isTypeRegistration(item) && !this.isFirm)
-    )
+    const conditions: Array<() => boolean> = []
+
+    // any cases not listed below are allowed (enabled)
+    conditions[0] = () => !this.isAllowed(AllowableActions.FILE_CORRECTION)
+    conditions[1] = () => item.availableOnPaperOnly
+    conditions[2] = () => item.isTypeStaff
+    conditions[3] = () => item.isFutureEffective
+    conditions[4] = () => this.isTypeAlteration(item)
+    conditions[5] = () => this.isTypeTransition(item)
+    conditions[6] = () => (this.isTypeIncorporationApplication(item) && !this.isBComp)
+    conditions[7] = () => (this.isTypeChangeOfRegistration(item) && !this.isFirm)
+    conditions[8] = () => (this.isTypeConversion(item) && !this.isFirm)
+    conditions[9] = () => (this.isTypeCorrection(item) && !this.isFirm && !this.isBComp)
+    conditions[10] = () => (this.isTypeRegistration(item) && !this.isFirm)
+
+    return conditions.some(condition => condition())
```

Comment:
This iterates over the functions and runs each of them until one returns True. (So it doesn't do more work than needed.)

I thought I'd structure this code like this to better line up with the unit test (which verifies each condition separately). It should also be easier to add conditions here. What do you think?

Reply:
It looks pretty clean.